client/state/selectors/utils/revisions/index.js

```diff
@@ -7,35 +7,34 @@ import { cloneDeep, get, identity } from 'lodash';
  * Internal dependencies
  */
 import decodeEntities from 'lib/post-normalizer/rule-decode-entities';
-import { normalizePostForDisplay } from 'state/posts/utils';
 
-function normalizeForEditing( revision ) {
-	if ( ! revision ) {
-		return null;
-	}
+const runMaybe = ( input, fn ) =>
+	get( fn( [ input ] ), 0, null );
 
-	return decodeEntities( cloneDeep( revision ) );
-}
+const normalizeForFields = ( fields ) => ( revision ) =>
+	runMaybe( revision, maybe => maybe
+		.filter( Boolean )
+		.map( cloneDeep )
+		.map( ( post ) => decodeEntities( post, fields ) )
+	);
 
 const NORMALIZER_MAPPING = {
-	display: normalizePostForDisplay,
-	editing: normalizeForEditing,
+	display: normalizeForFields( [ 'content', 'excerpt', 'title', 'site_name' ] ),
+	editing: normalizeForFields( [ 'excerpt', 'title', 'site_name' ] ),
 };
 
-export function hydrateRevision( state, revision ) {
-	if ( ! revision ) {
-		return revision;
-	}
-
+const injectAuthor = ( state ) => ( revision ) => {
 	const author = get( state.users.items, revision.author );
-	if ( ! author ) {
-		return revision;
-	}
+	return author
+		? { ...revision, author }
+		: revision;
+};
 
-	return {
-		...revision,
-		author,
-	};
+export function hydrateRevision( state, revision ) {
```

Comment:
that `runMaybe` is a pretty neat trick :) 

Only change I see is `hydrateRevision` will return `null` instead of `revision` if it's `false`-y. Given its current usage - in the selectors -, it would mean we have loaded `false`-y revisions in the state in the first place. There is no test for that behaviour since we're entering defensive programming territory and unit tests can pile up really fast :D

Reply:
Yeah, I expected that if we ever step into `if ( ! revision )` then we most likely just have a `null` that got in there somehow — since we expect an object, which is always truthy, can we just establish that we should discard anything else?

Reply:
(if not, we can make this equivalent to pre-refactor code by appending `|| revision` to the `runMaybe( … )` call)

Reply:
oh no, I think it's probably better to have it consistently return `null` in the long run, it makes for a more predictable interface :)

Sorry, I should have made it clearer that It's just something to keep in mind in case some dirty state gets in as there is nothing to defend against it in the [reducer](https://github.com/Automattic/wp-calypso/blob/2baa22cebd9c76a853b8e413b65a0ed2d9df0322/client/state/posts/revisions/reducer.js#L39) or in the [data layer](https://github.com/Automattic/wp-calypso/blob/2baa22cebd9c76a853b8e413b65a0ed2d9df0322/client/state/data-layer/wpcom/posts/revisions/index.js#L28)